### ISSUE TITLE
TST: failures in `test_riemann_zeta_complex` on windows in conda-forge

### ISSUE THREAD

**Issue by h-vetinari** [scipy/scipy#21927]
Due to another issue, rather than waiting for the rc's, I've run some [builds](https://github.com/conda-forge/scipy-feedstock/pull/284) in conda-forge from main. As of today, this leads to test failures on windows:
```
=========================== short test summary info ===========================
FAILED special/tests/test_zeta.py::test_riemann_zeta_complex[(-217.40285743524163+13.992648136816397j)-(-6.012818500554211e+249-1.926943776932387e+250j)-5e-13] - AssertionError: 
Not equal to tolerance rtol=5e-13, atol=0

nan location mismatch:
 ACTUAL: array(nan+nanj)
 DESIRED: array(-6.012819e+249-1.926944e+250j)
FAILED special/tests/test_zeta.py::test_riemann_zeta_complex[(-237.71710702931668+13.992648136816397j)-(-8.823803086106129e+281-5.009074181335139e+281j)-1e-13] - AssertionError: 
Not equal to tolerance rtol=1e-13, atol=0

nan location mismatch:
 ACTUAL: array(nan+nanj)
 DESIRED: array(-8.823803e+281-5.009074e+281j)
FAILED special/tests/test_zeta.py::test_riemann_zeta_complex[(-217.40285743524163-16.147667799398363j)-(-5.111612904844256e+251-4.907132127666742e+250j)-5e-13] - AssertionError: 
Not equal to tolerance rtol=5e-13, atol=0

nan location mismatch:
 ACTUAL: array(nan+nanj)
 DESIRED: array(-5.111613e+251-4.907132e+250j)
FAILED special/tests/test_zeta.py::test_riemann_zeta_complex[(-237.71710702931668-16.147667799398363j)-(-1.3256112779883167e+283-2.253002003455494e+283j)-5e-13] - AssertionError: 
Not equal to tolerance rtol=5e-13, atol=0

nan location mismatch:
 ACTUAL: array(nan+nanj)
 DESIRED: array(-1.325611e+283-2.253002e+283j)
= 4 failed, 72607 passed, 4723 skipped, 326 xfailed, 16 xpassed, 2594 warnings in 845.67s (0:14:05) =
```

CC @steppi for https://github.com/scipy/scipy/pull/21744

**Comment by rgommers** [comment]
This should be addressed for 1.15.0 or we're going to run into it again. `test_riemann_zeta_complex` is already marked as skipped for Intel compilers, and it seems like with Clang-cl on Windows they also don't work. 

@steppi I propose to move the 4 problematic underflow test cases to a separate test (same as for overflow in `test_riemann_zeta_avoid_overflow`), and mark that as running only with GCC on Linux and Clang on macOS. The rest is essentially fragile and uninteresting I think, it's going to fail on some other niche platforms as well.

**Comment by rgommers** [comment]
For the record, the 4 very large values in the issue description all do indeed underflow, as do 2 others, I noticed from the warnings escaping (that will be fixed) in parallel testing on macOS arm64 with Clang:
```
% pixi r test-nogil -s special -- --parallel-threads=4
....
FAILED scipy/special/tests/test_zeta.py::test_riemann_zeta_complex[(-217.40285743524163+13.992648136816397j)-(-6.012818500554211e+249-1.926943776932387e+250j)-5e-13] - scipy.special._sf_error.SpecialFunctionError: scipy.special/_riemann_zeta: (underflow) floating point underflow
FAILED scipy/special/tests/test_zeta.py::test_riemann_zeta_complex[(-237.71710702931668+13.992648136816397j)-(-8.823803086106129e+281-5.009074181335139e+281j)-1e-13] - scipy.special._sf_error.SpecialFunctionError: scipy.special/_riemann_zeta: (underflow) floating point underflow
FAILED scipy/special/tests/test_zeta.py::test_riemann_zeta_complex[(-294.86605461349745+13.992648136816397j)-(nan+infj)-1e-13] - scipy.special._sf_error.SpecialFunctionError: scipy.special/_riemann_zeta: (underflow) floating point underflow
FAILED scipy/special/tests/test_zeta.py::test_riemann_zeta_complex[(-217.40285743524163-16.147667799398363j)-(-5.111612904844256e+251-4.907132127666742e+250j)-5e-13] - scipy.special._sf_error.SpecialFunctionError: scipy.special/_riemann_zeta: (underflow) floating point underflow
FAILED scipy/special/tests/test_zeta.py::test_riemann_zeta_complex[(-237.71710702931668-16.147667799398363j)-(-1.3256112779883167e+283-2.253002003455494e+283j)-5e-13] - scipy.special._sf_error.SpecialFunctionError: scipy.special/_riemann_zeta: (underflow) floating point underflow
FAILED scipy/special/tests/test_zeta.py::test_riemann_zeta_complex[(-294.86605461349745-16.147667799398363j)-(nan-infj)-1e-13] - scipy.special._sf_error.SpecialFunctionError: scipy.special/_riemann_zeta: (underflow) floating point underflow
```

**Comment by steppi** [comment]
> @steppi I propose to move the 4 problematic underflow test cases to a separate test (same as for overflow in `test_riemann_zeta_avoid_overflow`), and mark that as running only with GCC on Linux and Clang on macOS. The rest is essentially fragile and uninteresting I think, it's going to fail on some other niche platforms as well.

Sounds good. I'll put a PR together now.
